src/sentry/lang/native/plugin.py

```diff
@@ -96,7 +96,11 @@ def inject_apple_backtrace(data, frames, diagnosis=None, error=None,
         exc = exception_from_apple_error_or_diagnosis(error, diagnosis)
         if exc is not None:
             exc['stacktrace'] = stacktrace
-            data['sentry.interfaces.Exception'] = exc
+            data['sentry.interfaces.Exception'] = {'values': [exc]}
+            # Since we inject the exception late we need to make sure that
+            # we set the event type to error as it would be set to
+            # 'default' otherwise.
+            data['type'] = 'error'
```

Comment:
Is this a sensible default value here? I'm not fully sure of the context, but this will get rendered oddly in the UI as just an "error" prefix to everything. Not sure if you have a better option though.